keras/engine/topology.py

```diff
@@ -2831,7 +2834,7 @@ def preprocess_weights_for_loading(layer, weights,
                 weights[0] = np.transpose(weights[0], (2, 3, 0, 1))
 
         if layer.__class__.__name__ == 'Conv3D':
-            if layer.data_format == 'channels_first':
+            if layer.data_format == 'channels_first' and original_backend in {'th', 'theano'}:
```

Comment:
Maybe specifying `original_dim_ordering` would be better, but I was trying to use the arguments present and it seemed like `dim_ordering` and the backend were more coupled in Keras 1 so I thought it would be ok.